Neos.Flow/Classes/Security/Aspect/LoggingAspect.php

```diff
@@ -81,17 +85,26 @@ public function logManagerLogout(JoinPointInterface $joinPoint)
         if (!$securityContext->isInitialized()) {
             return;
         }
-        $accountIdentifiers = [];
-        foreach ($securityContext->getAuthenticationTokens() as $token) {
-            /** @var $account Account */
-            $account = $token->getAccount();
-            if ($account !== null) {
-                $accountIdentifiers[] = $account->getAccountIdentifier();
-            }
-        }
+
+        $accountIdentifiers = array_reduce($securityContext->getAuthenticationTokens(), [$this, 'reduceTokenToAccountIdentifier'], []);
```

Comment:
Is there a specific reason why you use `array_reduce` here (that needs to copy & mutate the array for every iteration?) `array_map` would be the better fit here IMO. Besides (nitpick) do we really need the protected `reduceTokenToAccountIdentifier` method? It seems to be only needed here so I'd prefer a closure or at least a private method.
This could work and is not less comprehensible IMO:

```
$accountIdentifiers = [];
array_walk($securityContext->getAuthenticationTokens(), function(TokenInterface $token) use (&$accountIdentifiers) {
    $account = $token->getAccount();
    if ($account === null) {
        return;
    }
    $accountIdentifiers[] = $token-> getAccountIdentifier();
});
```

Reply:
I guess it's a matter of personal preference. I detest reference and array_walk, for me they are smells and easily lead to side effects. array_reduce is a purer way of manipulation IMHO. For map/filter to work you would need to map, filter and map again, and I wasn't exactly a fan of that as it felt to be too many operations, but you are right that this would be even nicer. I could change that though if you like.

Reply:
OK, let's try to take personal preference out of the game: Out of curiosity I wrote a little benchmark and while the `array_walk` approach is a little faster a good old `foreach` is more than twice as fast and IMO even easier to read:
```
$accountIdentifiers = [];
foreach ($securityContext->getAuthenticationTokens() as $token) {
    $account = $token->getAccount();
    if ($account !== null) {
        $accountIdentifiers[] = $account->getAccountIdentifier();
    }
}
```
You suggested that in an unrelated change (you know where *g) and I think it just makes sense here, as well :)